test/common_tests.dart

```diff
@@ -2188,7 +2188,7 @@ void runCommonTests(
             isSinkClosed = true;
           });
 
-          test('throwsIfEncodingIsNullAndWriteObject', () {
+          test('throwsIfEncodingIsNullAndWriteObject', () async {
             sink.encoding = null;
             expect(() => sink.write('Hello world'), throwsNoSuchMethodError);
```

Comment:
I'm curious why this is needed...

I should be able to say `expect(new Future.error('foo'), throwsA(anything));` without the test method needing to be async...